src/lib/transition_router/initial_validator.ml

```diff
@@ -228,6 +228,13 @@ let run ~logger ~trust_system ~verifier ~transition_reader
                ~rejected_blocks_logger ~time_received duplicate_checker logger
                transition_with_hash ;
              let sender = Envelope.Incoming.sender transition_env in
+             let blockchain_length =
+               External_transition.consensus_state transition_with_hash.data
+               |> Consensus.Data.Consensus_state.blockchain_length
+               |> Mina_numbers.Length.to_int
+             in
+             Mina_metrics.Transition_frontier
+             .update_max_unvalidated_blocklength_observed blockchain_length ;
```

Comment:
should we update this before the `if` statement on line 219, logging expired blocks as well?